manifests/install.pp

```diff
@@ -2,10 +2,38 @@
 #
 # This class is called from puppet_agent for install.
 #
-class puppet_agent::install {
+# === Parameters
+#
+# [package_file_name]
+#   The puppet-agent package file name.
+#   (see puppet_agent::prepare::package_file_name)
+#
+class puppet_agent::install(
+  $package_file_name = undef,
+) {
   assert_private()
 
-  package { $::puppet_agent::package_name:
-    ensure => present,
+  if $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '10' {
+    if $puppet_agent::params::master_agent_version != $::aio_agent_version {
+
+      contain puppet_agent::install::remove_packages
+
+      exec { 'replace puppet.conf removed by package removal':
```

Comment:
@ericwilliamson so the puppet package provider is running:

/bin/rpm -i --replacepkgs /opt/puppetlabs/packages/puppet-agent-1.2.5-1.sles10.x86_64.rpm

(--replacepkgs is there because I added it into install_options parameters).  This still results in conflicts with the existing pe-puppet.  I believe the provider is only going to run rpm -U if it detects an existing puppet-agent version.  

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/package/rpm.rb#L114-L128 

So, we could use an Exec to rpm -U, or continue to uninstall prior to install for 3.8.x sles10/el4/aix upgrades.  I'm leaning towards the later.

Thoughts?